Formula/libobjc2.rb

```diff
@@ -0,0 +1,57 @@
+class Libobjc2 < Formula
+  desc "Objective-C runtime library intended for use with Clang"
+  homepage "https://github.com/gnustep/libobjc2"
+  url "https://github.com/gnustep/libobjc2/archive/refs/tags/v2.1.tar.gz"
+  sha256 "78fc3711db14bf863040ae98f7bdca08f41623ebeaf7efaea7dd49a38b5f054c"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :test
+  # Clang explicitly forbids building Mach-O binaries of libobjc2.
+  # https://reviews.llvm.org/D46052
+  # macOS provides an equivalent Objective-C runtime.
+  depends_on :linux
+
+  uses_from_macos "llvm" => [:build, :test]
+
+  # Clang must be used on Linux because GCC Objective-C support is insufficient.
+  fails_with :gcc
+
+  resource "robin-map" do
+    url "https://github.com/Tessil/robin-map/archive/refs/tags/v1.0.1.tar.gz"
+    sha256 "b2ffdb623727cea852a66bddcb7fa6d938538a82b40e48294bb581fe086ef005"
+  end
+
+  def install
+    (buildpath/"third_party/robin-map").install resource("robin-map")
+
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    pkgshare.install "Test"
+  end
+
+  test do
+    # ENV.cc returns llvm_clang, which does not work in a test block.
+    ENV["CC"] = Formula["llvm"].opt_bin/"clang"
+
+    # Copy over test library and header and runtime test.
+    cp pkgshare/"Test/Test.h", testpath/"Test.h"
+    cp pkgshare/"Test/Test.m", testpath/"Test.m"
+    cp pkgshare/"Test/RuntimeTest.m", testpath/"RuntimeTest.m"
+
+    # Change Objective-C header path which assumes tests are being run in source tree.
+    inreplace testpath/"Test.h", "../objc", "objc"
+    inreplace testpath/"Test.m", "../objc", "objc"
```

Comment:
```suggestion
    inreplace ["Test.h", "Test.m"], "../objc", "objc"
```
I wonder if we should be doing the `inreplace` at `install` instead.